### PR TITLE
Changed log level to trace for connection recycle

### PR DESCRIPTION
**Pull request by christopinka**
Hello bikeshedder

Is this feasible? Is there a reason to keep this as INFO?

### CODE REVIEW

**Comment by christopinka** [comment]
@corynguyenfl  can you speak to why we hit this message repeatedly?

**Comment by corynguyenfl** [comment]
I think we hit repeatedly because the connection (tokio_postgres) is tied to runtime and is closed at the end of a runtime block.

File: postgres/src/lib.rs [hunk 1]
```diff
@@ -131,14 +131,14 @@ impl managed::Manager for Manager {
...
-            log::info!(target: "deadpool.postgres", "Connection could not be recycled: Connection closed");
+            log::trace!(target: "deadpool.postgres", "Connection could not be recycled: Connection closed");
             return Err(RecycleError::StaticMessage("Connection closed"));
         }
         match self.config.recycling_method.query() {
             Some(sql) => match client.simple_query(sql).await {
                 Ok(_) => Ok(()),
                 Err(e) => {
-                    log::info!(target: "deadpool.postgres", "Connection could not be recycled: {}", e);
+                    log::trace!(target: "deadpool.postgres", "Connection could not be recycled: {}", e);
```
[the remark this answers is not in the document]
Comment (christopinka) [reply postgres/src/lib.rs:141]: WARN works

**Comment by christopinka** [comment]
I think we can close this. We found an issue in our code with repeatedly creating tokio runtime. Unless you think 'WARN' is more appropriate per @abrown7100 's recommendation. Thanks

**Comment by bikeshedder** [comment]
I don't have a definite opinion on that. Maybe the core shouldn't even emit this warning but leave it to the actual implementation. e.g. the PostgreSQL pool might log a warning if the connection was terminated due to a timeout but log an info message if the connection was terminated because the database was shut down and it was a "proper" disconnect.

I don't have a definite opinion on which log level is correct if it's part of the core.

`WARN` makes sense if connections aren't expected to die and should just keep working.
`INFO` makes sense if it is to be expected but should rarely happen.
`DEBUG` or `TRACE` only makes sense if it happens a lot and we don't worry about it at all.

I'm torn between `WARN` and `INFO` ...

**Comment by bikeshedder** [comment]
After some consideration I'm changing this to `WARN`. Connections closed by the DB are indeed unexpected and should create a warning, I think.

**Pull request closed without merging**